src/Decorators/PipelineDecorator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Lorisleiva\Actions\Decorators;
+
+use Closure;
+use Exception;
+use Lorisleiva\Actions\Concerns\DecorateActions;
+
+class PipelineDecorator
+{
+    use DecorateActions;
+
+    public function __construct($action)
+    {
+        $this->setAction($action);
+    }
+
+    public function __invoke(mixed ...$arguments): mixed
+    {
+        return $this->handleFromAnyMethod(...$arguments);
+    }
+
+    public function handle(mixed ...$arguments): mixed
+    {
+        return $this->handleFromAnyMethod(...$arguments);
+    }
+
+    /**
+     * Typical pipeline behavior expects two things:
+     *
+     *     1)  The pipe class to expect a single incoming parameter (along with
+     *         a closure) and single return value.
+     *     2)  The pipe class to be aware of the next closure and determine what
+     *         should be passed into the next pipe.
+     *
+     * Because of these expectations, this behavior is asserting two opinions:
+     *
+     *     1)  Regardless of the number of parameters provided to the asPipeline
+     *         method implemented here, only the first will be supplied to the
+     *         invoked Action.
+     *     2)  If the invoked Action does not return anything, then the next
+     *         closure will be supplied the same parameter. However, if the
+     *         invoked action does return a non-null value, that value will
+     *         be supplied to the next closure.
+     */
+    protected function handleFromAnyMethod(mixed ...$arguments): mixed
+    {
+        $passable = array_shift($arguments);
+        $closure = array_pop($arguments);
+        $returned = null;
+
+        if ($this->hasMethod('asPipeline')) {
+            $returned = $this->callMethod('asPipeline', [$passable]);
```

Comment:
100% personal preference here but I'd prefer:

```php
$returned = $this->callMethod('asPipeline', [$passable]);
return $closure($returned ?? $passable);
```

Here to remove the `else`

Reply:
There is a test case for the following scenario:

> it can run as a noop/passthrough pipe in a pipeline, without a handle or asPipeline method 

This implementation causes this test case to fail. 

So, I think the main question is - is this a valid test case to support? Do we care if the Action class does not have a handle or asPipeline method? Should it throw an exception and break the pipeline if the Action class doesn't have either method? or should we simply pass the $passable into the $next closure and not break anything?

Reply:
If we want to keep that test case and not expect an exception, the following (or similar) could work a bit more cleanly:

```php
        $method = match (true) {
            $this->hasMethod('asPipeline') => 'asPipeline',
            $this->hasMethod('handle') => 'handle',
            default => null
        };

        if ($method) {
            return $closure($this->callMethod($method, [$passable]) ?? $passable);
        }

        return $closure($passable);
```

Reply:
What's the use case for using an Action as a pipeline that does nothing unconditionally?